ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,513 @@
+      Roles:
+        - !Ref 'InstanceRole'
+
+  # The EC2 instance writes to it, but this is introduced for the cleanup of the parameter
+  InstanceSSMParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      # This will need to be synced with the `packer-rstudio` project
+      Name: !Sub "/service-catalog/synapse/cred/${LinuxInstance}/oidc-accesstoken"
+      Type: String
+      Value: to be replaced by EC2
+
+  SSMParameterEncryptionKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: key used to encrypt SSM Parameters for a EC2 instance
+      # number of days to wait before key gets deleted
+      PendingWindowInDays: 7
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
+        # https://dev.to/jansonsa/aws-how-to-deny-access-to-resources-while-allowing-a-specific-role-547b
+        - Sid: "Deny decrypt to all Principals that shouldn't require access"
+          Effect: 'Deny'
+          Principal: "*"
+          Condition:
+            StringNotEquals:
+              "aws:PrincipalArn": !Sub "${InstanceRole.Arn}"
+          Action:
+            - kms:Decrypt
+          Resource: '*'
+        - Sid: Allow use of the key
+          Effect: Allow
+          Principal:
+            AWS: !Sub "${InstanceRole.Arn}"
+          Action:
+          - kms:DescribeKey
+          - kms:Encrypt
+          - kms:Decrypt
+          - kms:ReEncrypt*
+          - kms:GenerateDataKey
+          - kms:GenerateDataKeyWithoutPlaintext
+          Resource: '*'
+        - Sid: 'Default KMS Policy that enables IAM permissions'  # https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
+          Effect: 'Allow'
+          Principal:
+            AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+          Action:
+            - 'kms:*'
+          Resource: '*'
+        - Sid: "Allow administration of the key to CFN service role"
+          Effect: "Allow"
+          Principal:
+            AWS:
+              - !ImportValue
+                'Fn::Sub': '${AWS::Region}-sc-ec2vpc-launchrole-LaunchRoleArn'
+          Action:
+            - "kms:Create*"
+            - "kms:Describe*"
+            - "kms:Enable*"
+            - "kms:List*"
+            - "kms:Put*"
+            - "kms:Update*"
+            - "kms:Revoke*"
+            - "kms:Disable*"
+            - "kms:Get*"
+            - "kms:Delete*"
+            - "kms:ScheduleKeyDeletion"
+            - "kms:CancelKeyDeletion"
+          Resource: "*"
+
+  SSMParameterEncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      # This will need to be synced with the `packer-rstudio` project
+      AliasName: !Sub 'alias/service-catalog/synapse/cred/${LinuxInstance}'
+      TargetKeyId: !Ref SSMParameterEncryptionKey
+
+  LinuxInstance:
+    Type: AWS::EC2::Instance
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E1022
+            - W1011
+      'AWS::CloudFormation::Init':
```

Comment:
In the current SC deployment of rstudio we also [setup synapse credentials for apache and rstudio](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/templates/ec2/sc-ec2-linux-jumpcloud-notebook.yaml#L299-L333) and I believe those things depend on [set_env_vars](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/templates/ec2/sc-ec2-linux-jumpcloud-notebook.yaml#L285-L299).  I think we still need to do that somewhere in here?

Reply:
Yes, you are right.  The environment variables, `SYNAPSE_TOKEN_AWS_SSM_PARAMETER_NAME` and `AWS_DEFAULT_REGION` have to be set, to allow the Synapse client to retrieve the Synapse access token from parameter store. (See [this](https://raw.githubusercontent.com/Sage-Bionetworks/service-catalog-utils/v2.0.3/linux/opt/sage/bin/rstudio_synapse_cred_env.sh) script.).  I have updated this PR to add the two env var's to both the RStudio and Jupyter notebooks.